### PR TITLE
Replace types for the FQCN's

### PR DESCRIPTION
**Pull request by ZRazor**
I am targetting this branch, because it's BC.

```markdown
### Changed
- Replaced types for the FQCN's
```

### CODE REVIEW

File: src/BasketBundle/Form/AddressType.php [hunk 1]
```diff
@@ -51,6 +51,19 @@ public function __construct($addressClass, BasketInterface $basket)
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $choiceType = 'Symfony\Component\Form\Extension\Core\Type\ChoiceType';
```
Comment (OskarStark) [inline src/BasketBundle/Form/AddressType.php]: can you move choice below entity type, to keep the same order like in the form?

File: src/BasketBundle/Form/AddressType.php [hunk 2]
```diff
@@ -51,6 +51,19 @@ public function __construct($addressClass, BasketInterface $basket)
...
+            $entityType = 'Symfony\Bridge\Doctrine\Form\Type\EntityType';
+            $countryType = 'Symfony\Component\Form\Extension\Core\Type\CountryType';
+            $submitType = 'Symfony\Component\Form\Extension\Core\Type\SubmitType';
+        } else {
+            $choiceType = 'choice';
```
Comment (OskarStark) [inline src/BasketBundle/Form/AddressType.php]: same here

File: src/CustomerBundle/Admin/AddressAdmin.php [hunk 3]
```diff
@@ -34,6 +34,17 @@ public function configure()
      */
     public function configureFormFields(FormMapper $formMapper)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $countryType = 'Symfony\Component\Form\Extension\Core\Type\CountryType';
```
Comment (OskarStark) [inline src/CustomerBundle/Admin/AddressAdmin.php]: same here, can you please use the same order as use in the formMapper?

File: src/CustomerBundle/Admin/AddressAdmin.php [hunk 4]
```diff
@@ -92,9 +110,16 @@ public function configureListFields(ListMapper $list)
      */
     protected function configureDatagridFilters(DatagridMapper $filter)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $addressTypeType = 'Sonata\CustomerBundle\Form\Type\AddressTypeType';
+        } else {
+            $addressTypeType = 'sonata_customer_address_types';
```
Comment (OskarStark) [inline src/CustomerBundle/Admin/AddressAdmin.php]:
```diff
- $addressTypeType
+ $addressType
```

File: src/CustomerBundle/Admin/AddressAdmin.php [hunk 5]
```diff
@@ -92,9 +110,16 @@ public function configureListFields(ListMapper $list)
...
+        }
+
         $filter
             ->add('current')
-            ->add('type', null, array(), 'sonata_customer_address_types', array('translation_domain' => 'SonataCustomerBundle'))
+            ->add('type', null, array(), $addressTypeType, array('translation_domain' => 'SonataCustomerBundle'))
```
Comment (OskarStark) [inline src/CustomerBundle/Admin/AddressAdmin.php]: same here

File: src/CustomerBundle/Admin/CustomerAdmin.php [hunk 6]
```diff
@@ -34,17 +34,28 @@ public function configure()
      */
     public function configureFormFields(FormMapper $formMapper)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $modelListType = 'Sonata\AdminBundle\Form\Type\ModelListType';
```
Comment (OskarStark) [inline src/CustomerBundle/Admin/CustomerAdmin.php]: please use the same order as its used in the formMapper

File: src/InvoiceBundle/Admin/InvoiceAdmin.php [hunk 7]
```diff
@@ -45,19 +45,32 @@ public function configure()
      */
     public function configureFormFields(FormMapper $formMapper)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $countryType = 'Symfony\Component\Form\Extension\Core\Type\CountryType';
```
Comment (OskarStark) [inline src/InvoiceBundle/Admin/InvoiceAdmin.php]: please use the same order as its used in the FormMapper

File: src/InvoiceBundle/Admin/InvoiceAdmin.php [hunk 8]
```diff
@@ -82,12 +95,21 @@ public function configureFormFields(FormMapper $formMapper)
      */
     public function configureListFields(ListMapper $list)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $currencyType = 'Sonata\Component\Currency\CurrencyFormType';
```
Comment (OskarStark) [inline src/InvoiceBundle/Admin/InvoiceAdmin.php]: same order as in the FormMapper please

File: src/OrderBundle/Admin/OrderAdmin.php [hunk 9]
```diff
@@ -77,6 +77,27 @@ public function configure()
      */
     public function configureFormFields(FormMapper $formMapper)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $countryType = 'Symfony\Component\Form\Extension\Core\Type\CountryType';
```
Comment (OskarStark) [inline src/OrderBundle/Admin/OrderAdmin.php]: order please

File: src/OrderBundle/Admin/OrderAdmin.php [hunk 10]
```diff
@@ -133,6 +154,15 @@ public function configureFormFields(FormMapper $formMapper)
      */
     public function configureListFields(ListMapper $list)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $currencyType = 'Sonata\Component\Currency\CurrencyFormType';
```
Comment (OskarStark) [inline src/OrderBundle/Admin/OrderAdmin.php]: order please

File: src/OrderBundle/Admin/OrderAdmin.php [hunk 11]
```diff
@@ -156,6 +186,17 @@ public function configureListFields(ListMapper $list)
      */
     public function configureDatagridFilters(DatagridMapper $filter)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $orderStatusType = 'Sonata\OrderBundle\Form\Type\OrderStatusType';
```
Comment (OskarStark) [inline src/OrderBundle/Admin/OrderAdmin.php]: order please

File: src/ProductBundle/Admin/DeliveryAdmin.php [hunk 12]
```diff
@@ -35,17 +35,28 @@ public function configure()
      */
     public function configureFormFields(FormMapper $formMapper)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $countryType = 'Symfony\Component\Form\Extension\Core\Type\CountryType';
```
Comment (OskarStark) [inline src/ProductBundle/Admin/DeliveryAdmin.php]: order please

File: src/ProductBundle/Admin/ProductAdmin.php [hunk 13]
```diff
@@ -135,12 +135,21 @@ public function getPersistentParameters()
      */
     public function configureListFields(ListMapper $list)
     {
+        // NEXT_MAJOR: Keep FQCN when bumping Symfony requirement to 2.8+.
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            $currencyType = 'Sonata\Component\Currency\CurrencyFormType';
```
Comment (OskarStark) [inline src/ProductBundle/Admin/ProductAdmin.php]: order please

File: src/BasketBundle/Form/AddressType.php (hunk 1 above)
Comment (ZRazor) [inline src/BasketBundle/Form/AddressType.php]: I get it

File: src/CustomerBundle/Admin/AddressAdmin.php (hunk 4 above)
Comment (ZRazor) [inline src/CustomerBundle/Admin/AddressAdmin.php]: There are two address types in this bundle: `Sonata\CustomerBundle\Form\Type\AddressTypeType` and`Sonata\CustomerBundle\Form\Type\AddressType`
Thats why I named this variable `addressTypeType`
Maybe name it `addressTypesType` ?

Comment (ZRazor) [inline src/CustomerBundle/Admin/AddressAdmin.php]: IMO `AddressTypeType` is ugly, we should rename it in the next major ;)

Comment (OskarStark) [inline src/CustomerBundle/Admin/AddressAdmin.php]: oh youre right 👍

**Comment by OskarStark** [comment]
Thank you @ZRazor 👍

**Pull request merged**